src/common/deco.rs

```diff
@@ -312,11 +332,24 @@ impl Deco {
     fn deco_stop_depth(&self, ceiling: Depth) -> Depth {
         DEFAULT_CEILING_WINDOW * (ceiling / DEFAULT_CEILING_WINDOW).ceil()
     }
+
+    fn validate_gas_mixes<T: DecoModel>(&self, deco_model: &T, gas_mixes: &[Gas]) -> Result<(), DecoCalculationError> {
```

Comment:
You are not using self. It feels like validation is a static thing and could be external to this I think?

Reply:
Or am I missing somehting and &self is actually needed ? :P